src/Spawner.cpp

```diff
@@ -145,8 +145,16 @@ Spawner::ProcessHandle::ProcessHandle(Deployment *deploment, bool redirectOutput
         {
             throw std::runtime_error("Error, log directory '" + logDir + "' does not exist, but it should !");
         }
-        processName = deploment->getName();
-        redirectOutput(logDir + "/" + processName + "-" + boost::lexical_cast<std::string>(getpid()) + ".txt");
+        processName = deployment->getName();
+
+        if (!textLogFileName.empty())
+        {
+            redirectOutput(logDir + "/" + textLogFileName + "-" + boost::lexical_cast<std::string>(getpid()) + ".txt");
```

Comment:
Now we have a varibable of the function called `textLogFileName`, but it will be modified such that it is actually not the resulting file name. This is counter intuitive.

To deal with this we could either
1. document the behavior correctly and rename things
2. shift the logic of taking care that no files are overwritten to the caller

I think the most intuitive and usual way is 2. I.e. keep the behavior as is when no log file name is given and if one is given assume that the caller knows what it did when passing the name.

For the caller we're looking at (orogen-execution) asimilar mechanism as for the the data logs could be introduced, where the value of a counter varibale is attached (cf. https://git.hb.dfki.de/dfki-control/modeling/cnd-execution/-/blob/master/src/LogManager.cpp#L238)

Reply:
Just to clarify: My suggestion was to keep the `textLogFileName` argument, but treat it as the name implies.. the name of the log file. And only shift the responsibility to give a reasonable name to the caller (if the caller wants to make use if this optional argument)

Reply:
Thanks for the feedback. Yes, it would be confusing for the client to define a given name for the file and then random numbers come at the end of it in the actual file name.